samples/chaincode/cc-tools-demo/README.md

```diff
@@ -0,0 +1,334 @@
+
+## Clone the cc-tools-demo chaincode
+
+We need to clone the chaincode folder from the [cc-tools-demo](https://github.com/hyperledger-labs/cc-tools-demo) repository here.
+Note: In some cases, git commands need sudo permissions
+
+```bash
+export ccToolsDemoPath=$FPC_PATH/samples/chaincode/cc-tools-demo
+sudo git clone -n --no-checkout --depth=1 --filter=tree:0 https://github.com/hyperledger-labs/cc-tools-demo.git "$ccToolsDemoPath/chaincode"
+cd "$ccToolsDemoPath/chaincode" || { echo "$ccToolsDemoPath/chaincode does not exist" ; exit 1; }
+sudo chown -R $USER $ccToolsDemoPath
+git config --global --add safe.directory /src/github.com/hyperledger/fabric-private-chaincode/samples/chaincode/cc-tools-demo/chaincode
+git sparse-checkout set --no-cone chaincode/*
+git checkout
+mv chaincode/* $ccToolsDemoPath
+cd $ccToolsDemoPath
+sudo rm -r $ccToolsDemoPath/chaincode
+```
+
+The chaincode code structure is different than normal chaincode as it's using the cc-tools framework.
+
+## Edit the chaincode to became an FPC chaincode instead of normal fabric
+
+Go to `$FPC_PATH/samples/chaincode/cc-tools-demo/main.go` and view the project structure.
+
+```bash
+cd $FPC_PATH/samples/chaincode/cc-tools-demo
+```
+
+The code presented here is not different from traditional Go Chaincode developed with CC-Tools framework. All the FPC specific protection mechanisms are handled by the FPC framework transparently.
+
+The `main.go` contains the starting point of the chaincode.
+
+To use FPC, we need to add some logic to instantiate our private chaincode and start it.
+To do so, we use `shim.Start(fpc.NewPrivateChaincode(new(CCDemo)))` and since we're already in the FPC repo we will point to the local FPC package.
+
+Go to the `go.mod` file in the `cc-tools-demo` chaincode folder and add the following replace line before the `require( )` block:
+
+```go
+replace github.com/hyperledger/fabric-private-chaincode => ../../../ 
+```
+
+CC-tools-demo chaincode has its own packages that are needed, so we run `go mod tidy` in the `cc-tools-demo` folder.
+
+Then, edit the chaincode in the `$FPC_PATH/samples/chaincode/cc-tools-demo/main.go` file:
+
+For import block, add the fpc package
+
+```go
+import (
+	///Keep everything as is
+	fpc "github.com/hyperledger/fabric-private-chaincode/ecc_go/chaincode"
+)
+```
+
+In the `main()` function, replace the following `if` statement:
+
+```go
+if os.Getenv("RUN_CCAAS") == "true" {
+	err = runCCaaS()
+} else {
+	err = shim.Start(new(CCDemo))
+}
+
+```
+
+With this:
+
+```go
+if os.Getenv("RUN_CCAAS") == "true" {
+	err = runCCaaS()
+} else {
+	if os.Getenv("FPC_ENABLED") == "true" {
+		err = shim.Start(fpc.NewPrivateChaincode(new(CCDemo)))
+	} else {
+		err = shim.Start(new(CCDemo))
+	}
+}
+```
+
+In the `runCCaaS()` function, replace the following lines:
+
+```go
+ccid := os.Getenv("CHAINCODE_ID") //This needs to be replaced
+
+server := &shim.ChaincodeServer{
```

Comment:
there is code missing between ccid and server ... either separate both replacements or add the missing code to both snippets

Reply:
Done